src/components/home/Matches.js

```diff
@@ -1,10 +1,61 @@
-import React from 'react';
+import React, { Fragment, Component } from 'react';
+import { connect } from 'react-redux';
+import { array, func } from 'prop-types';
 import { FormattedMessage } from 'react-intl';
+import { getMatches } from 'actions/matchesActions';
+import UserAvatar from 'resources/icons/UserAvatar';
 
-const Matches = () => (
-  <h2 className="subtitle center">
-    <FormattedMessage id="home.sidebar.noMatches" />
-  </h2>
-);
+class Matches extends Component {
+  componentDidMount() {
+    const { matches, getMatches } = this.props;
 
-export default Matches;
+    !matches && getMatches();
```

Comment:
To unify behavior, I think it's better to initialize this array as an empty one in the reducer's initial state. That way, you avoid doing these null/undefined controls and you just map over what you already now is an array. 

Only thing it can be empty, which is absolutely fine if you haven't fetched any matches yet or simply the user doesn't have any.